backend/api.test/Controllers/RobotModelControllerTests.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Api.Database.Models;
+using Api.Services;
+using Api.Test.Database;
+using Microsoft.Extensions.DependencyInjection;
+using Testcontainers.PostgreSql;
+using Xunit;
+
+namespace Api.Test.Controllers;
+
+public class RobotModelControllerTests : IAsyncLifetime
+{
+    public required DatabaseUtilities DatabaseUtilities;
+    public required PostgreSqlContainer Container;
+    public required HttpClient Client;
+    public required JsonSerializerOptions SerializerOptions;
+
+    public required IRobotModelService RobotModelService;
+
+    public async Task InitializeAsync()
+    {
+        (Container, var connectionString, var connection) =
+            await TestSetupHelpers.ConfigurePostgreSqlDatabase();
+        var factory = TestSetupHelpers.ConfigureWebApplicationFactory(
+            postgreSqlConnectionString: connectionString
+        );
+        var serviceProvider = TestSetupHelpers.ConfigureServiceProvider(factory);
+
+        Client = TestSetupHelpers.ConfigureHttpClient(factory);
+        SerializerOptions = TestSetupHelpers.ConfigureJsonSerializerOptions();
+
+        DatabaseUtilities = new DatabaseUtilities(
+            TestSetupHelpers.ConfigurePostgreSqlContext(connectionString)
+        );
+
+        RobotModelService = serviceProvider.GetRequiredService<IRobotModelService>();
+    }
+
+    public Task DisposeAsync() => Task.CompletedTask;
+
+    [Fact]
+    public async Task CheckThatListAllRobotModelsReturnsSuccess()
+    {
+        var response = await Client.GetAsync("/robot-models");
+        var robotModels = await response.Content.ReadFromJsonAsync<List<RobotModel>>(
+            SerializerOptions
+        );
+
+        // Seven models are added by default to the database
+        // This number must be changed if new robots are introduced
+        Assert.Equal(7, robotModels!.Count);
```

Comment:
Would it maybe be good to have a clean database for the tests in this file and then add the models here using DatabaseUtilities? It's not too clear where these 7 are added at the momnt. I'm OK with the tests as is, this is just a thought.